process/sync/baseSync.go

```diff
@@ -154,67 +157,49 @@ func (boot *baseBootstrap) processReceivedHeader(headerHandler data.HeaderHandle
 		log.Debug("forkDetector.AddHeader", "error", err.Error())
 	}
 
-	boot.mutRcvHdrHash.Lock()
-	hash := boot.requestedHeaderHash()
-	if hash == nil {
-		boot.mutRcvHdrHash.Unlock()
-		return
-	}
+	go boot.requestMiniBlocks(headerHandler.GetShardID(), headerHandler.GetNonce())
```

Comment:
We should now change this method. You should send headerHash instead shardId and in its implementation (shardBootstrap/metaBootstrap) you should replace GetMetaHeaderFromPoolWithNonce/GetShardHeaderFromPoolWithNonce with GetMetaHeaderFromPool/GetShardHeaderFromPool as we can now go targeted

Reply:
done.